test/compiler.test.js

```diff
@@ -439,6 +439,86 @@ describe('compiler', function() {
+        bootScripts: ['custom-boot/init.js']
+      });
+      expect(instructions.files.boot).to.eql([initJs]);
+    });
+
+    it('should resolve non-relative path in `bootDirs`', function() {
+      appdir.createConfigFilesSync();
+      var initJs = appdir.writeFileSync('custom-boot/init.js',
+        'module.exports = function(app) { app.fnCalled = true; };');
+      var instructions = boot.compile({
+        appRootDir: appdir.PATH,
+        bootDirs:['custom-boot']
+      });
+      expect(instructions.files.boot).to.eql([initJs]);
+    });
+
+    it('resolves missing extensions in `bootScripts`', function() {
+      appdir.createConfigFilesSync();
+      var initJs = appdir.writeFileSync('custom-boot/init.js',
+        'module.exports = function(app) { app.fnCalled = true; };');
+      var instructions = boot.compile({
+        appRootDir: appdir.PATH,
+        bootScripts:['./custom-boot/init']
+      });
+      expect(instructions.files.boot).to.eql([initJs]);
+    });
+
+    it('explores index.js in `bootDirs`', function() {
+      appdir.createConfigFilesSync();
+      var initJs = appdir.writeFileSync('custom-boot/index.js',
+        'module.exports = function(app) { app.fnCalled = true; };');
+      var instructions = boot.compile({
+        appRootDir: appdir.PATH,
+        bootDirs:['./custom-boot']
+      });
+      expect(instructions.files.boot).to.eql([initJs]);
+    });
+
+    it('resolves module relative path for `bootScripts`', function() {
+      appdir.createConfigFilesSync();
+      var initJs = appdir.writeFileSync('node_modules/custom-boot/init.js',
+        'module.exports = function(app) { app.fnCalled = true; };');
+      var instructions = boot.compile({
+        appRootDir: appdir.PATH,
+        bootScripts: ['custom-boot/init.js']
+      });
+      expect(instructions.files.boot).to.eql([initJs]);
+    });
+
+    it('resolves module relative path in `bootScripts`', function() {
+      appdir.createConfigFilesSync();
+      var initJs = appdir.writeFileSync('node_modules/custom-boot/init.js',
+        'module.exports = function(app) { app.fnCalled = true; };');
+      var instructions = boot.compile({
+        appRootDir: appdir.PATH,
+        bootScripts: ['custom-boot/init.js']
+      });
+      expect(instructions.files.boot).to.eql([initJs]);
+    });
+
+    it('explores `bootScripts` in app relative path', function() {
+      appdir.createConfigFilesSync();
+      var appJs = appdir.writeFileSync('./custom-boot/init.js',
+        'module.exports = function(app) { app.fnCalled = true; };');
+      var moduleJs = appdir.writeFileSync('node_modules/custom-boot/init.js',
+        'module.exports = function(app) { app.fnCalled = true; };');
+      var instructions = boot.compile({
+        appRootDir: appdir.PATH,
+        bootScripts: ['custom-boot/init.js']
+      });
+      expect(instructions.files.boot).to.not.eql([moduleJs]);
+      expect(instructions.files.boot).to.eql([appJs]);
```

Comment:
Please keep only the second assertion (expectation) line. When the array has only a single item "appJs", then it cannot be an array with a single item "moduleJs", so it's not needed to test that.